examples/hosted-delivery-function-templates/javascript/build-delivery-functions.js

```diff
@@ -0,0 +1,87 @@
+const esbuild = require('esbuild');
+const { join, parse, resolve } = require('path');
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers');
+
+const manifest = require('./contentful-app-manifest.json');
+
+const argv = yargs(hideBin(process.argv)).argv;
+
+const validateActions = () => {
+  const requiredProperties = ['id', 'path', 'entryFile'];
+  const uniqueValues = new Set();
+
+  if (manifest.deliveryFunctions.length > 1) {
+    throw new Error(
+      `The maximum amount of delivery functions is 1, found ${manifest.deliveryFunctions.length}`
+    );
+  }
+
+  manifest.deliveryFunctions.forEach((deliveryFunction) => {
+    requiredProperties.forEach((property) => {
+      if (!deliveryFunction.hasOwnProperty(property)) {
+        throw new Error(
+          `Delivery function with name: '${deliveryFunction.name}' is missing the '${property}' property`
+        );
+      }
+    });
+
+    const { id, path, entryFile } = deliveryFunction;
+
+    if (uniqueValues.has(id)) {
```

Comment:
You were correct since we enforced at most 1 delivery function. I've since removed this check as we don't want to enforce this during build time but during deploy time. Now the `forEach` can run multiple times which could trigger the below errors if multiple delivery functions share the same properties.